### DISCUSSION TITLE
[zodios-openapi] More string formats

### DISCUSSION THREAD

**Discussion by epotter2297** [ecyrbe/zodios#224] [Ideas]
Right now if I have an endpoint with a parameter like this:
```js
{
  name: 'email',
  type: 'Query',
  schema: z.string().email(),
},
```
It shows up in OpenAPI with a type of `string` and an empty format.

According to the [official spec](https://swagger.io/specification/), things like `email` or `url` aren't official string formats, but it is an open field, so technically any string is valid. It would be nice if things like `z.string().email()` or `z.string().url()` had sane defaults and / or I could optionally add define my own custom format like this:
```js
{
  name: 'email',
  type: 'Query',
  schema: z.string().email(),
  format: 'email',
},
```

**Comment by epotter2297** [comment]
Another thought that just occurred to me, you wouldn't be able to specify a format for a property in the response object with this approach... we could still do default formats for things like `z.string().email()` though.

**Reply by ecyrbe** [reply]
Maybe i'm confused, but the generated schema is this one:
```json
{
   "type": "string",
    "format": "email"
}
```
So maybe it's just an issue with swagger-ui ?

**Reply by epotter2297** [reply]
It might have been, let me double check...

**Reply by epotter2297** [reply]
Ok yeah this one is just a mistake on my part, you can disregard this post then 🙂
